happiness-backend/tests/test_happiness.py

```diff
@@ -427,3 +427,43 @@ def test_discussion_comments(init_client):
     comments = comment_access_u3.json
     assert len(comments) == 2
     assert comments[0]['author']['id'] == 1 and comments[1]['author']['id'] == 3
+
+def test_discussion_comments_edit_delete(init_client):
+    client, tokens = init_client
+    client.post('/api/group/', json={'name': 'group 1'}, headers=auth_header(tokens[0]))
+    get_group_by_id(1).add_users(['user2'])
+    client.post('/api/happiness/', json={
+        'value': 4.5,
+        'comment': 'bad day',
+        'timestamp': '2023-06-19'
+    }, headers=auth_header(tokens[0]))
+    h= client.post('/api/happiness/1/comment', json={
+        'text': 'oh no what happened?'
+    }, headers=auth_header(tokens[1]))
+    client.post('/api/happiness/1/comment', json={
+        'text': 'call tonight?'
+    }, headers=auth_header(tokens[1]))
+
+    bad_edit1 = client.put('/api/happiness/comments/5', json={
+        'data': 'are you feeling ok?'
+    }, headers=auth_header(tokens[0]))
+    assert bad_edit1.status_code == 404
+
+    bad_edit2 = client.put('/api/happiness/comments/1', json={
+        'data': 'are you feeling ok?'
+    }, headers=auth_header(tokens[0]))
+    assert bad_edit2.status_code == 403
+
+    edit = client.put('/api/happiness/comments/1', json={
+        'data': 'are you feeling ok?'
+    }, headers=auth_header(tokens[1]))
+    assert edit.status_code == 200
+
+    delete = client.delete('/api/happiness/comments/1', headers=auth_header(tokens[1]))
+    assert delete.status_code == 204
```

Comment:
maybe add a "bad delete" test where you try to delete a comment that doesn't exist, or you try to delete a comment where it's unauthorized?